test/e2e/build-sample-external-plugin.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+# TODO: Dynamically set exteranl plugin destination based on OS platform
+EXTERNAL_PLUGIN_DESTINATION_PREFIX="${HOME}/Library/Application Support/kubebuilder/plugins"
+PLUGIN_NAME="sampleexternalplugin"
+PLUGIN_VERSION="v1"
+EXTERNAL_PLUGIN_DESTINATION="${EXTERNAL_PLUGIN_DESTINATION_PREFIX}/${PLUGIN_NAME}/${PLUGIN_VERSION}"
+EXTERNAL_PLUGIN_PATH="${EXTERNAL_PLUGIN_DESTINATION}/${PLUGIN_NAME}"
+
+if [ -d "$EXTERNAL_PLUGIN_DESTINATION" ]; then
+  echo "$EXTERNAL_PLUGIN_DESTINATION does exist."
+  if [ -e "$EXTERNAL_PLUGIN_PATH" ]; then
+    echo "clean up old binary..."
+    rm "$EXTERNAL_PLUGIN_PATH" 
+  fi
+else
+    mkdir -p "$EXTERNAL_PLUGIN_DESTINATION"
+fi
+
+SOURCE_DIR="./docs/book/src/simple-external-plugin-tutorial/testdata/sampleexternalplugin/v1"
+cd $SOURCE_DIR && go build -o $PLUGIN_NAME && mv $PLUGIN_NAME "$EXTERNAL_PLUGIN_PATH"
+
+kubebuilder init --plugins="${PLUGIN_NAME}/${PLUGIN_VERSION}" --help | grep --quiet "$PLUGIN_NAME"
+
+if [ $? -eq 0 ]; then
+    echo "External plugin is built at: $EXTERNAL_PLUGIN_DESTINATION"
+else
+    echo "FAIL to build external plugin"
```

Comment:
It would be nice to echo the exact error message as well, with which `init` command has failed.

Reply:
+1

Reply:
IMO, ideally, we need to keep all in Go.
Why? It makes it easier for us to keep the project maintained.
- See that using an IDE, we can trigger the tests directly, which means the sh would not be executed.
- Also, it makes it more intuitive since all the required steps to do the tests are placed together.
- By leasing, keeping it in Go with the test help us to troubleshot since we can add breakpoints to verify each step

Reply:
I agree it is a good practice  to follow the current pattern, which lets Ginkgo do all e2e related tasks, instead of having an additional script. By doing this we can keep consistency and benefit from Ginkgo's feature.

The current script may be a reference to put in the docs to provide users info about how to install their external plugin.